service/matching/taskWriter.go

```diff
@@ -196,31 +196,22 @@ func (w *taskWriter) appendTasks(
 ) (*persistence.CreateTasksResponse, error) {
 
 	resp, err := w.tlMgr.db.CreateTasks(tasks)
-	switch err.(type) {
-	case nil:
-		return resp, nil
-
-	case *persistence.ConditionFailedError:
-		w.tlMgr.signalFatalProblem(w.tlMgr)
-		return nil, err
-
-	default:
+	if err != nil {
+		w.tlMgr.signalIfFatal(err)
```

Comment:
This may be broken. Previously, if `db.CreateTasks` returned a transient storage error or some soft of context timeout, we would log it and return `nil, err`. Now going through `signalIfFatal`, if ResilientSyncMatch is off (the default) then we will tear down the taskQueueManager on such errors. Probably not the behavior we want but I'm interested to hear other perspectives.

Reply:
Sort of convinced myself this is okay. If we say that when Resilient sync-match is disabled then _any_ error from the persistence layer should cause the TQM to be unloaded that that's consistent with this code. Anyone have the historical insight to know if that is the intent of non-resilient matching error behavior? In other words, was the previous implementation actually incorrect? @yiminc

Reply:
We should change the default for resilient sync-match to true in 1.13

Reply:
Should I hold off on this until we're closer to 1.13 then?

Reply:
No, we should land sooner. Please also put up another PR to change the default config for resilient sync-match to true as well.